### PR TITLE
Refactor code according GitHub comments 

### PR DESCRIPTION
**Pull request by asaquzzaman**

### CODE REVIEW

File: pro/includes/class-foogallery-pro-exif.php [hunk 1]
```diff
@@ -16,6 +16,258 @@ function __construct() {
...
             //Add lightbox EXIF options
             add_filter( 'foogallery_build_container_attributes', array( $this, 'add_lightbox_data_attributes' ), 20, 2 );
+
+            if ( is_admin() ) {
+                //add extra fields to the templates that support exif
+                add_filter( 'foogallery_override_gallery_template_fields', array( $this, 'add_exif_fields' ), 20, 2 );
+            }
+
+            //set the settings icon for Exif
+            add_filter( 'foogallery_gallery_settings_metabox_section_icon', array( $this, 'add_section_icons' ) );
```
Comment (bradvin) [inline pro/includes/class-foogallery-pro-exif.php:26]: this only needs to run when is_admin = true

File: pro/includes/class-foogallery-pro-exif.php [hunk 2]
```diff
@@ -16,6 +16,258 @@ function __construct() {
...
+
+            //add some settings for EXIF
+            add_filter( 'foogallery_admin_settings_override', array( $this, 'add_exif_settings' ) );
```
Comment (bradvin) [inline pro/includes/class-foogallery-pro-exif.php:29]: this only needs to run when is_admin = true

File: pro/includes/class-foogallery-pro-exif.php [hunk 3]
```diff
@@ -14,11 +14,8 @@ function __construct() {
             //Add EXIF data
             add_filter( 'foogallery_attachment_html_link_attributes', array( $this, 'add_exif' ), 10, 3 );
             
-            //Add container class
-            add_filter( 'foogallery_build_class_attribute', array( $this, 'add_container_class' ), 10,  2 );
```
Comment (bradvin) [inline pro/includes/class-foogallery-pro-exif.php:18]: why did you remove this hook? It was correct

File: pro/includes/class-foogallery-pro-exif.php [hunk 4]
```diff
@@ -41,20 +39,6 @@ function add_lightbox_data_attributes( $attributes, $gallery ) {
...
-        /**
-         * Customize the item container class
-         *
-         * @param $classes
-         * @param $gallery
-         * 
-         * @return array
-         */
-        function add_container_class( $classes, $gallery ) {
-            $classes[] = 'fg-exif-top-right fg-exif-dark fg-exif-rounded';
```
Comment (bradvin) [inline pro/includes/class-foogallery-pro-exif.php:53]: This function was correct, and should be added back again
Check if EXIF is enabled and add class names from the other EXIF gallery settings
Instead of using hard-coded class names, get the gallery setting values using function foogallery_gallery_template_setting.

File: pro/includes/class-foogallery-pro-exif.php [hunk 5]
```diff
@@ -41,20 +39,6 @@ function add_lightbox_data_attributes( $attributes, $gallery ) {
             return $attributes;
```
Comment (bradvin) [inline pro/includes/class-foogallery-pro-exif.php:291]: This is where you can add the global languages settings, eg. Aperture Text and Camera Text

File: pro/includes/class-foogallery-pro-exif.php [hunk 6]
```diff
@@ -65,6 +49,12 @@ function add_container_class( $classes, $gallery ) {
...
         function add_exif( $attr, $args, $foogallery_attachment ) {
+            global $current_foogallery;
+
+            if ( $current_foogallery->lightbox != 'foogallery' ) {
+                return $attr;
+            }
+
         	$meta = wp_get_attachment_metadata( $foogallery_attachment->ID ); 
 
 			$attr['data-exif'] = json_encode( $meta['image_meta'] );
```
Comment (bradvin) [inline pro/includes/class-foogallery-pro-exif.php:312]: This is where you should filter the keys in $meta to make sure only the keys allowed in Allowed EXIF Attributes global setting are included.

File: pro/includes/class-foogallery-pro-exif.php (hunk 4 above)
Comment (asaquzzaman) [reply pro/includes/class-foogallery-pro-exif.php:53]: Include in new commit

File: pro/includes/class-foogallery-pro-exif.php (hunk 5 above)
Comment (asaquzzaman) [reply pro/includes/class-foogallery-pro-exif.php:343]: Would you please mention the data attributes where i please the global language settings. I am doing this, but it's not working.
```
$attributes['data-foogallery-lightbox']->il8n = json_encode( array (
      'exif' => array(
               'aperture' =>  'Text from settings'
      )
) )
```

Comment (bradvin) [reply pro/includes/class-foogallery-pro-exif.php:343]: I will test this out and let you know
